### PR TITLE
Self-host abstract Collection protocol in pure Beamtalk (BT-815)

### PR DESCRIPTION
**Pull request by jamesc**
## Summary

Implements Phase 2b of ADR 0034 (Stdlib Self-Hosting): replaces `@primitive` bodies on Collection's abstract protocol methods with pure Beamtalk implementations.

- **8 methods self-hosted**: `includes:`, `detect:`, `detect:ifNone:`, `anySatisfy:`, `allSatisfy:`, `collect:`, `select:`, `reject:` — all implemented in pure BT using `do:` with non-local return (`^`) or delegation to `inject:into:`
- **1 method kept as primitive**: `inject:into:` remains `@primitive` because the compiler's mutation threading doesn't work through abstract-class `do:` calls (generates `lists:foreach` instead of `lists:foldl`)
- **Argument order fix**: `beamtalk_collection_ops:inject_into/3` now wraps the block to call `Block(Acc, Elem)`, matching the Beamtalk `block value: acc value: each` convention
- **7 Erlang functions removed** from `beamtalk_collection_ops.erl` (only `inject_into/3` and `to_list/1` remain)
- **New test fixture**: `NumberList` — user-defined Collection subclass implementing only `do:`, `size`, `printString`
- **11 new tests** in `custom_collection_test.bt` including non-commutative inject and empty collection edge cases

### Follow-up
- BT-820: List `inject:into:` has pre-existing argument order inconsistency with Collection convention

Closes https://linear.app/beamtalk/issue/BT-815

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive test coverage for custom collection implementations, including filtering, transformation, and iteration methods.
  * Introduced test fixture for user-defined collection subclasses.

* **Refactor**
  * Streamlined collection operations implementation for improved maintainability.

### CODE REVIEW

File: stdlib/test/custom_collection_test.bt [hunk 1]
```diff
@@ -0,0 +1,68 @@
+// Copyright 2026 James Casey
+// SPDX-License-Identifier: Apache-2.0
+
+// BT-815: Tests that a user-defined Collection subclass (implementing only
+// do: and size) correctly inherits the abstract Collection protocol.
+
+TestCase subclass: CustomCollectionTest
```
[the remark this answers is not in the document]
Comment (jamesc) [reply stdlib/test/custom_collection_test.bt]: No fix needed — the `// load` directive is not required here. The Beamtalk test runner auto-discovers and compiles all files under `stdlib/test/fixtures/` as part of the test bundle. Other fixture-dependent tests like `instance_class_var_access_value_type_test.bt` (which uses `ClassVarPoint` from a fixture) also omit `// load` directives and pass correctly. The tests already pass without this directive (391/391 passing). The `// load` pattern in `remove_from_system_test.bt` is a special case for explicit ordering of a parent/child class hierarchy, not a general requirement.
